lib/oli_web/live/projects/projects_live.ex

```diff
@@ -37,7 +37,8 @@ defmodule OliWeb.Projects.ProjectsLive do
 
   data author, :any
   data is_admin, :boolean, default: false
-  data changeset, :any, default: Project.changeset(%Project{title: ""})
+
+  data changeset, :any, default: Project.new_project_changeset(%Project{title: ""})
```

Comment:
I had a problem here, and I don't think this is the best way to do this, but wasn't sure how else to solve it.

As soon as I added the cast_embed to the normal Project.changest, I started getting this compile-time error on this line:

```
== Compilation error in file lib/oli_web/live/projects/projects_live.ex ==
** (ArgumentError) cannot escape #Function<19.44846763/2 in Ecto.Changeset.on_cast_default/2>. The supported values are: lists, tuples, maps, atoms, numbers, bitstrings, PIDs and remote functions in the format &Mod.fun/arity
    (elixir 1.13.2) src/elixir_quote.erl:185: :elixir_quote.argument_error/1
    (elixir 1.13.2) src/elixir_quote.erl:340: :elixir_quote.do_quote/3
    (elixir 1.13.2) src/elixir_quote.erl:471: :elixir_quote.do_quote_simple_list/4
    (elixir 1.13.2) src/elixir_quote.erl:471: :elixir_quote.do_quote_simple_list/4
    (elixir 1.13.2) src/elixir_quote.erl:386: :elixir_quote.do_escape/3
    (elixir 1.13.2) src/elixir_quote.erl:340: :elixir_quote.do_quote/3
...
```

My elixir knowledge hit it's limit here and I'm not sure how to properly solve that. So instead, I made two different changeset functions, one with the embed, and one without (since this didn't need to actually set any of those embedded attributes)

Reply:
Im not entirely sure, but I think if you used the original changeset and supplied a default value for attributes like this, that it would work:

```
data changeset, :any, default: Project.new_project_changeset(%Project{title: "", attributes: %{learning_language: "en"})
```

Reply:
I tried several variations (including exactly that) on that idea with no luck.

I don't think having a separate changeset for creating a project here is a horrible solution, it just doesn't feel necessary.

Reply:
Agreed.  After it lands as-is perhaps we can figure out a way to remove the need for it.